### ISSUE TITLE
Google Domains

### ISSUE THREAD

**Issue by celenityy** [hagezi/dns-blocklists#2900]
### Which domain(s) should be blocked?

auditrecording-pa.googleapis.com
clienttracing-pa.googleapis.com
datasaver.googleapis.com
feedback-pa.googleapis.com
growth-pa.googleapis.com
gvt2-cn.com
gvt3.com
locationhistory-pa.googleapis.com
locationhistoryaggregates-pa.googleapis.com

### Why should these domain(s) be blocked?

`auditrecording-pa.googleapis.com` - It looks like this is already blocked on `Ultimate`, but I think it might be worth blocking this on some of the less aggressive lists as well. This appears to be related to Google's Cloud Audit Logging API: https://cloud.google.com/logging/docs/audit, so it's just more telemetry. I was also able to find more info on it proving this further [here](https://www.scss.tcd.ie/Doug.Leith/pubs/apple_google_additional_material.pdf) on Page 34 & Page 36. We can see: `POST https://auditrecording−pa.googleapis.com/google.internal.api.auditrecording.v1.AuditRecordingMobileService/CreateAuditRecord
POST body:
<binary payload>` & `POST https://auditrecording−pa.googleapis.com/google.internal.api.
auditrecording.v1.AuditRecordingMobileService/CreateAuditRecord
Headers
x−goog−spatula: CjYKFmNvbS5nb29n...Fwu6/s+CCz+wBKgA=
POST body decoded as protobuf:
<...>
<device details>
<...>`.

`clienttracing-pa.googleapis.com` - I've noticed Stock Android devices phone home to this, difficult to find info on, but it seems to be related to Google's [Cloud Trace API](https://cloud.google.com/trace/docs/reference/v2/rest/), so just more telemetry.

`datasaver.googleapis.com` - This is related to Chromium's "Data Saver" feature, I think we should really consider blocking this one, at least on the more aggressive lists. You can see [here](https://www.fastvue.co/sophos/blog/google-data-saver-affect-security-confidentiality-reporting/) for more info, but this is **extremely** concerning from a privacy & security standpoint... Google is not only sent the websites you visit, but literally MITMs them.

`feedback-pa.googleapis.com` - I've noticed Google apps on iOS and Android frequently call this domain, I wasn't able to find any official references or documentation on it, but it appears to be used for telemetry based off this [Czech forum post](https://forum.viry.cz/viewtopic.php?p=1552134&sid=78c6f5d12d61596281ce3bd0c0e135c0#p1552134) - `annotation=anonymous_feedback_submit_apicka_apkey=AIzaSejQEszExuVHsgk8YElDTlbbrhzeRISc --anenation=anonymous_feedback_submit_url=https://feedback-pa.googleapis.com/v1/feedback/products/1633926/web:anonymous_submit`

`growth-pa.googleapis.com` - I saw this domain was requested before in #44 & #2411, and it seems like it wasn't blocked due to the lack of information. However, I was able to find info on it, and it appears this domain is directly related to & used for promotions in Google's apps & services. See [here](https://chromium.googlesource.com/chromium/src/+/c26ff44a53..bf7ad46dee) from Chromium's source code, & [here](https://www.scss.tcd.ie/doug.leith/apple_google.pdf) on Page 9:

`-const char kDesktopIOSPromotionQueryPhoneNumber[] =`
`-    "https://growth-pa.googleapis.com/v1/get_verified_phone_numbers";`
`-`
`-const char kDesktopIOSPromotionSendSMS[] =`
`-    "https://growth-pa.googleapis.com/v1/send_sms";`
`-`

`The Google Docs and Messaging apps (or a process on its behalf) connect to growth-pa.googleapis.com/google.internal.identity.growth.v1.GrowthApiService/GetPromos, sending device details but no unique identifiers`

`gvt2-cn.com` - Chinese variant of `gvt2.com`, which we already block, used for telemetry.

`gvt3.com` - Similar to `gvt2.com`, [according to Google, this is used for "Google Play connectivity monitoring and diagnostics"](https://support.google.com/work/android/answer/10513641).

`locationhistory-pa.googleapis.com` - This is extremely difficult to find information on, but it appears to be related to Google's location history. I think it'd probably make sense to block this on the more aggressive tests. This might need more investigation though, unclear.

`locationhistoryaggregates-pa.googleapis.com` - Same as above.

**Comment by hagezi** [comment]
Thank you @Retold3202, great work as always. We'll take a closer look.

@xRuffKez

**Comment by hagezi** [comment]
Thanks for the details @Retold3202 - great. I have added `feedback-pa.clients6.google.com`.

I will not completely block Google Maps Timeline through `timeline.google.com`, the data collection is blocked in Ultimate by blocking userlocation.googleapis.com.

**Comment by hagezi** [comment]
@Retold3202 Can you confirm that `geller-pa.googleapis.com` needs to be unblocked for the Google Assistant to work? It is currently unblocked.

There is also `device-provisioning.googleapis.com`, which also appears frequently in the logs. I once blocked it as a test and found that the push notifications could no longer be configured in the F1TV iOS app as a result.

**Comment by hagezi** [comment]
Many thanks @Retold3202, `geller-pa.googleapis.com` added to aggressive lists.

**Comment by Retold3202** [comment]
> I had personally blocked voilatile-pa.googleapis.com in the past but I then whitelisted it because, if I remember right, after some days Google Play Store showed "Device not certified" under Play Protect Certification

I've had it blocked on my device for around a week now, and I just took a look, and it appears to still show as "Certified". So it might've just been a coincidence, or maybe it's something different in my case. Some of these Google domains are probably the hardest I've ever tried researching/finding info on, a lot of them are fairly sketchy IMO and there's near no info anywhere on a lot of them unfortunately.

> content-autofill.googleapis.com

I'm fairly certain this is related to the "Autofill with Google" feature on Android. Personally I could care less about it and I block it on my personal list, but I doubt Hagezi will want to block this for the masses since this could be a legitimate feature people are using.

> lamssettings-pa.googleapis.com

This appears to be related to geolocation (ex. Page 3 [here](https://www.scss.tcd.ie/doug.leith/pubs/additional_material_neversleepingears.pdf)). As to what it's purpose is... your guess is as good as mine. 🤷‍♀️ I've had it blocked and haven't noticed any issues, looks like you have as well, but certainly needs more investigation.

> infinitedata-pa.googleapis.com

This is definitely a sketchy one, I've ran into it myself and spent a lot of time researching it. After some more research, according to Page 53 [here](https://www.scss.tcd.ie/doug.leith/pubs/additional_material_neversleepingears.pdf)), this appears to be another form of telemetry & was caught here sending the list of installed apps to Google. I haven't encountered any issues with this blocked, and & you said you haven't either, so I think we should probably consider blocking this on the lists.

> quake-pa.googleapis.com

Another domain with absolutely no info on it anywhere. Definitely needs investigation. 

> chromesyncentities-pa.googleapis.com

This is probably related to Google Chrome Sync. This is another case where I doubt Hagezi will block it for the masses since it could be a legitimate feature that people are using. I'll probably add it to my personal blocklist though because I could care less about it.

> nearbysharing-pa.googleapis.com

This is definitely another sketchy one. It appears to be required for Google's ["Quick Share"](https://support.google.com/android/answer/9286773) feature to work (ex. [here](https://github.com/safing/portmaster/issues/1454)), but [this does also appear to be used for advertising & tracking](https://developers.google.com/android/reference/com/google/android/gms/nearby/connection/package-summary). I think we should block this on the more aggressive lists.

> remoteprovisioning.googleapis.com

This shouldn't be blocked, it's part of Android's [Remote Provisioning Attestation](https://android-developers.googleblog.com/2022/03/upgrading-android-attestation-remote.html) feature, which is useful from a security perspective & used for checking integrity.

**Comment by Retold3202** [comment]
Also appears I was 100% correct about `footprints-pa.googleapis.com`, this is definitely used for Google's Web & Activity History. See Page 18 & 42 [here](https://www.scss.tcd.ie/doug.leith/pubs/additional_material_neversleepingears.pdf).

`POST https://footprints−pa.googleapis.com/footprints.oneplatform.
FootprintsService/GetActivityControlsSettings`

**Comment by Retold3202** [comment]
As an aside, I think we should also consider blocking:

`admob.googleapis.com` - Google [AdMob API](https://developers.google.com/admob/api/v1/reference/rest/).

`adsensehost.googleapis.com` - Google [Adsense "Host" API](https://console.cloud.google.com/marketplace/product/google/adsensehost.googleapis.com).

`audit.googleapis.com` - Google [Cloud Audit Logging](https://cloud.google.com/logging/docs/audit).

`analyticsadmin.googleapis.com` - Google ["Analytics Admin" API](https://console.cloud.google.com/marketplace/product/google/analyticsadmin.googleapis.com).

`analyticsdata.googleapis.com` - Google ["Analytics Data" API](https://developers.google.com/analytics/devguides/reporting/data/v1/rest/).

`analyticshub.googleapis.com` - Google ["Analytics Hub" API](https://cloud.google.com/bigquery/docs/reference/analytics-hub/rest/).

`analyticsreporting.googleapis.com` - Google ["Analytics Reporting" API](https://developers.google.com/analytics/devguides/reporting/core/v4/rest/).

`appenginereporting.googleapis.com` - Google ["App Engine Reporting Service"](https://console.cloud.google.com/apis/library/appenginereporting.googleapis.com).

`chromeuxreport.googleapis.com` - Google ["Chrome UX Report" API](https://developers.google.com/web/tools/chrome-user-experience-report/api/reference/rest/).

`cloudaudit.googleapis.com` - Same as `audit.googleapis.com`, also used for Cloud Audit Logging.

`cloudlatencytest.googleapis.com` - Google "[Cloud Network Performance Monitoring" API](https://console.cloud.google.com/marketplace/product/google/cloudlatencytest.googleapis.com).

`cloudprofiler.googleapis.com` - Google [Cloud Profiler Observability & Monitoring](https://cloud.google.com/profiler/docs).

`cloudtrace.googleapis.com` - Google's ["Cloud Trace" API](https://cloud.google.com/trace/docs/reference/v2/rest/).

`content-partnersbadge-pa.googleapis.com` - Google ["Partner Badge Program"](https://support.google.com/google-ads/answer/9028798) - Just more marketing & tracking.

`dfareporting.googleapis.com` - Google Doubleclick ["Campaign Manager 360" API](https://developers.google.com/doubleclick-advertisers/rest/).

`doubleclickbidmanager.googleapis.com` - Google Doubleclick ["Bid Manager" API](https://developers.google.com/bid-manager/reference/rest/).

`doubleclicksearch.googleapis.com` - Google Doubleclick ["Search Ads 360" API](https://console.cloud.google.com/marketplace/product/google/doubleclicksearch.googleapis.com).

`federatedcompute-pa.googleapis.com` - Part of Google's "Private Compute Services" - ["Enables privacy-preserving aggregate machine learning and analytics across many devices, without any raw data leaving the device](https://github.com/google/private-compute-services?tab=readme-ov-file#current-apis). 

`firebasepredictions.googleapis.com` - Google Firebase "Predictions", just more creepy ad/tracking... See [here](https://firebase.google.com/downloads/Halfbrick_Case_Study.pdf).

`firewallinsights.googleapis.com` - Google ["Firewall Insights"](https://cloud.google.com/network-intelligence-center/docs/firewall-insights/concepts/overview).

`gapromomanager-pa.googleapis.com` - Not extremely clear, but appears to be another domain related to promotions in the "Google" app on Android, similar to `xgapromomanager-pa.googleapis.com`.

`logging.googleapis.com` - Google ["Cloud Logging" API](https://cloud.google.com/logging/docs/reference/v2/rest/).

`marketingplatformadmin.googleapis.com` - Google ["Marketing Platform Admin" API](https://developers.google.com/marketing-platform/devguides/api/admin/v1/rest).

`meshtelemetry.googleapis.com` - Google ["Mesh Telemetry & Monitoring" API](https://cloud.google.com/service-mesh/docs/observability-overview).

`mobilenetworkscoring-pa.googleapis.com` - This appears to be reporting sensitive network data back to Google in order to determine the "network quality". See Page 18 & 42 [here](https://www.scss.tcd.ie/doug.leith/pubs/additional_material_neversleepingears.pdf).

`monitoring.googleapis.com` - Google's ["Cloud Monitoring" API](https://cloud.google.com/monitoring/api/ref_v3/rest/).

`pagespeedmobilizer.com` - Google's ["Page Speed Insights" API](https://developers.google.com/speed/docs/insights/v5/about).

`pagespeedonline.googleapis.com` - Same as above.

`partners-json.googleapis.com` - Google "Partners API" - ["Lets advertisers search certified companies and create contact leads with them, and also audits the usage of clients"](https://console.cloud.google.com/marketplace/product/google/partners-json.googleapis.com).

`recommendationengine.googleapis.com` - Google ["Recommendations AI"](https://cloud.google.com/recommendations) & [here](https://cloud.google.com/recommendations-ai/docs/reference/rest/) - "Take advantage of Google's expertise in recommendations, powered by state-of-the-art machine learning models. Provide effective and real-time personalization with the recommendations capability (formerly Recommendations AI) of Vertex AI Search."

`recommender.googleapis.com` - Google ["Recommender"](https://cloud.google.com/recommender/docs) & [here](https://cloud.google.com/recommender/docs/reference/rest/). - "Recommender is a service on Google Cloud that provides usage recommendations and insights for Cloud products and services".

`stackdriver.googleapis.com` - Google's "Stackdriver" API - ["Used by Google Cloud’s operations suite to collect signals across Google Cloud internal and external apps, platforms, and services"](https://console.cloud.google.com/apis/library/stackdriver.googleapis.com).

`surveys.googleapis.com` - Google ["Surveys" API](https://cloud.google.com/blog/products/gcp/google-consumer-surveys-api-lets-you-add-a-survey-to-any-mobile-or-web-app).

`tagmanager.googleapis.com` - Google [Tag Manager API](https://console.cloud.google.com/marketplace/product/google/tagmanager.googleapis.com).

`timeseriesinsights.googleapis.com` - Google ["Time Series Insights" Monitoring API](https://cloud.google.com/timeseries-insights/docs/reference/rest).

`wirelessdevicestats.googleapis.com` - [Appears to be telemetry & data collection from Google Home](https://www.reddit.com/r/googlehome/comments/uu7bz2/wifi_sends_a_lot_of_data_even_though_i_opted_out/).

**Comment by hagezi** [comment]
In https://github.com/hagezi/dns-blocklists/commit/426849232c606444d5d3c78f7fd9be517ed9458b I have made a few adjustments based on the comments. Thnaks @AleIlMagno and @Retold3202.
[AleIlMagno](https://github.com/AleIlMagno), if you want the "full program", then I recommend using Ultimate.

**Comment by Retold3202** [comment]
Another domain I noticed devices on my network connecting to that might be worth blocking: `tracedepot-pa.googleapis.com` - Difficult to find info on, but its likely also related to Google's ["Cloud Trace"](https://cloud.google.com/trace/docs/reference/v2/rest/). I haven't seen any issues with it blocked.

**Comment by botollama** [comment]
* auditrecording-pa.googleapis.com likely relates to Google's internal audit logging or recording processes.
 * clienttracing-pa.googleapis.com might be involved in tracing or monitoring client interactions with Google services.
 * datasaver.googleapis.com could be associated with data saving or optimization features within Google products.
 * feedback-pa.googleapis.com might be used for collecting user feedback or reports.
 * growth-pa.googleapis.com could be related to internal tools or services that Google uses to track or analyze growth metrics.
 * gvt2-cn.com and gvt3.com are not part of the [invalid URL removed] domain and their purposes are unclear.
 * locationhistory-pa.googleapis.com and locationhistoryaggregates-pa.googleapis.com are likely connected to Google's location tracking features and data aggregation.
I asked ai gemma2:27b and this was the answer i got.

**Comment by botollama** [comment]
Adding with this i already have asked to block this in a previous request but that have not been merged but i have set that into my custom deny list could not find any such issues with my devices.

**Comment by hagezi** [comment]
Currently blocked in the individual list versions:
```
ADBLOCK  light.txt                    ||analytics.googleapis.com^
ADBLOCK  light.txt                    ||androidtvchannels-pa.googleapis.com^
ADBLOCK  light.txt                    ||clouderrorreporting.googleapis.com^
ADBLOCK  multi.txt                    ||ad-creatives-public.commondatastorage.googleapis.com^
ADBLOCK  multi.txt                    ||analytics.googleapis.com^
ADBLOCK  multi.txt                    ||androidtvchannels-pa.googleapis.com^
ADBLOCK  multi.txt                    ||clientmetrics-pa.googleapis.com^
ADBLOCK  multi.txt                    ||clouderrorreporting.googleapis.com^
ADBLOCK  pro.txt                      ||ad-creatives-public.commondatastorage.googleapis.com^
ADBLOCK  pro.txt                      ||analyticsadmin.googleapis.com^
ADBLOCK  pro.txt                      ||analyticsdata.googleapis.com^
ADBLOCK  pro.txt                      ||analytics.googleapis.com^
ADBLOCK  pro.txt                      ||analyticshub.googleapis.com^
ADBLOCK  pro.txt                      ||analyticsinsights-pa.googleapis.com^
ADBLOCK  pro.txt                      ||analyticsmobile-pa.googleapis.com^
ADBLOCK  pro.txt                      ||analyticsreporting.googleapis.com^
ADBLOCK  pro.txt                      ||analyticssearch-pa.googleapis.com^
ADBLOCK  pro.txt                      ||androidtvchannels-pa.googleapis.com^
ADBLOCK  pro.txt                      ||androidtvwatsonfe-pa.googleapis.com^
ADBLOCK  pro.txt                      ||auditrecording-pa.googleapis.com^
ADBLOCK  pro.txt                      ||clientmetrics-pa.googleapis.com^
ADBLOCK  pro.txt                      ||clienttracing-pa.googleapis.com^
ADBLOCK  pro.txt                      ||clouderrorreporting.googleapis.com^
ADBLOCK  pro.txt                      ||content-analyticsadmin.googleapis.com^
ADBLOCK  pro.txt                      ||content-analyticsdata.googleapis.com^
ADBLOCK  pro.txt                      ||content-analyticsreporting.googleapis.com^
ADBLOCK  pro.txt                      ||crashlyticsreports-pa.googleapis.com^
ADBLOCK  pro.txt                      ||firebaselogging.googleapis.com^
ADBLOCK  pro.txt                      ||firebaselogging-pa.googleapis.com^
ADBLOCK  pro.txt                      ||footprints-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||ad-creatives-public.commondatastorage.googleapis.com^
ADBLOCK  pro.plus.txt                 ||adexchangebuyer.googleapis.com^
ADBLOCK  pro.plus.txt                 ||adexchangebuyer-json.googleapis.com^
ADBLOCK  pro.plus.txt                 ||adexchange.googleapis.com^
ADBLOCK  pro.plus.txt                 ||adexchangeseller.googleapis.com^
ADBLOCK  pro.plus.txt                 ||admob.googleapis.com^
ADBLOCK  pro.plus.txt                 ||adsense.googleapis.com^
ADBLOCK  pro.plus.txt                 ||adsensehost.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analyticsadmin.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analyticsdata.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analytics.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analyticshub.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analyticsinsights-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analyticsmobile-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analyticsreporting.googleapis.com^
ADBLOCK  pro.plus.txt                 ||analyticssearch-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||androidtvchannels-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||androidtvwatsonfe-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||auditrecording-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||clientmetrics-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||clienttracing-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||clouderrorreporting.googleapis.com^
ADBLOCK  pro.plus.txt                 ||content-analyticsadmin.googleapis.com^
ADBLOCK  pro.plus.txt                 ||content-analyticsdata.googleapis.com^
ADBLOCK  pro.plus.txt                 ||content-analyticsreporting.googleapis.com^
ADBLOCK  pro.plus.txt                 ||crashlyticsreports-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||feedback-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||firebaselogging.googleapis.com^
ADBLOCK  pro.plus.txt                 ||firebaselogging-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||footprints-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||geller-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||growth-pa.googleapis.com^
ADBLOCK  pro.plus.txt                 ||optimizationguide-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||ad-creatives-public.commondatastorage.googleapis.com^
ADBLOCK  ultimate.txt                 ||adexchangebuyer.googleapis.com^
ADBLOCK  ultimate.txt                 ||adexchangebuyer-json.googleapis.com^
ADBLOCK  ultimate.txt                 ||adexchange.googleapis.com^
ADBLOCK  ultimate.txt                 ||adexchangeseller.googleapis.com^
ADBLOCK  ultimate.txt                 ||admob.googleapis.com^
ADBLOCK  ultimate.txt                 ||adsense.googleapis.com^
ADBLOCK  ultimate.txt                 ||adsensehost.googleapis.com^
ADBLOCK  ultimate.txt                 ||analyticsadmin.googleapis.com^
ADBLOCK  ultimate.txt                 ||analyticsdata.googleapis.com^
ADBLOCK  ultimate.txt                 ||analytics.googleapis.com^
ADBLOCK  ultimate.txt                 ||analyticshub.googleapis.com^
ADBLOCK  ultimate.txt                 ||analyticsinsights-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||analyticsmobile-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||analyticsreporting.googleapis.com^
ADBLOCK  ultimate.txt                 ||analyticssearch-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||androidtvchannels-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||androidtvwatsonfe-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||appenginereporting.googleapis.com^
ADBLOCK  ultimate.txt                 ||audit.googleapis.com^
ADBLOCK  ultimate.txt                 ||auditrecording-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||chromeuxreport.googleapis.com^
ADBLOCK  ultimate.txt                 ||clientmetrics-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||clienttracing-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||cloudaudit.googleapis.com^
ADBLOCK  ultimate.txt                 ||clouderrorreporting.googleapis.com^
ADBLOCK  ultimate.txt                 ||cloudlatencytest.googleapis.com^
ADBLOCK  ultimate.txt                 ||cloudprofiler.googleapis.com^
ADBLOCK  ultimate.txt                 ||cloudtrace.googleapis.com^
ADBLOCK  ultimate.txt                 ||content-analyticsadmin.googleapis.com^
ADBLOCK  ultimate.txt                 ||content-analyticsdata.googleapis.com^
ADBLOCK  ultimate.txt                 ||content-analyticsreporting.googleapis.com^
ADBLOCK  ultimate.txt                 ||content-partnersbadge-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||crashlyticsreports-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||datasaver.googleapis.com^
ADBLOCK  ultimate.txt                 ||dfareporting.googleapis.com^
ADBLOCK  ultimate.txt                 ||doubleclickbidmanager.googleapis.com^
ADBLOCK  ultimate.txt                 ||doubleclicksearch.googleapis.com^
ADBLOCK  ultimate.txt                 ||federatedcompute-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||feedback-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||firebaselogging.googleapis.com^
ADBLOCK  ultimate.txt                 ||firebaselogging-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||firebasepredictions.googleapis.com^
ADBLOCK  ultimate.txt                 ||firewallinsights.googleapis.com^
ADBLOCK  ultimate.txt                 ||footprints-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||gapromomanager-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||geller-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||geomobileservices-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||glsfrontend-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||growth-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||infinitedata-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||lamssettings-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||localservices.googleapis.com^
ADBLOCK  ultimate.txt                 ||locationhistoryaggregates-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||locationhistory-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||logging.googleapis.com^
ADBLOCK  ultimate.txt                 ||marketingplatformadmin.googleapis.com^
ADBLOCK  ultimate.txt                 ||meshtelemetry.googleapis.com^
ADBLOCK  ultimate.txt                 ||mobilecrashreporting.googleapis.com^
ADBLOCK  ultimate.txt                 ||mobilenetworkscoring-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||monitoring.googleapis.com^
ADBLOCK  ultimate.txt                 ||ogads-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||optimizationguide-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||pagespeedonline.googleapis.com^
ADBLOCK  ultimate.txt                 ||partners-json.googleapis.com^
ADBLOCK  ultimate.txt                 ||playmoviesdfe-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||recommendationengine.googleapis.com^
ADBLOCK  ultimate.txt                 ||recommender.googleapis.com^
ADBLOCK  ultimate.txt                 ||scone-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||searchlabspartnerservice-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||searchpromo-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||semanticlocation-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||stackdriver.googleapis.com^
ADBLOCK  ultimate.txt                 ||surveys.googleapis.com^
ADBLOCK  ultimate.txt                 ||tagmanager.googleapis.com^
ADBLOCK  ultimate.txt                 ||timeseriesinsights.googleapis.com^
ADBLOCK  ultimate.txt                 ||tracedepot-pa.googleapis.com^
ADBLOCK  ultimate.txt                 ||userlocation.googleapis.com^
ADBLOCK  ultimate.txt                 ||wirelessdevicestats.googleapis.com^
ADBLOCK  ultimate.txt                 ||xgapromomanager-pa.googleapis.com^
```

**Comment by user716253** [comment]
feedback-pa.clients6.google.com is needed for youtube "send feedback" feature.

**Comment by hagezi** [comment]
> feedback-pa.clients6.google.com

moved to ultimate.

**Comment by hagezi** [comment]
I found these weeks ago:
```
ogads-pa.googleapis.com
ogads-pa.clients6.google.com
```
It appears that this subdomains associated with Google's ad services infrastructure. This subdomains are likely used for handling requests related to advertisements, analytics, and other related services within the Google Ads ecosystem.

Currently only blocked in Ultimate, will move it up to Pro++

**Comment by hagezi** [comment]
@Retold3202 

![grafik](https://github.com/user-attachments/assets/ec38c8e6-382b-40ec-80b0-df7952d7e961)

There is a message that blocking `optimizationguide-pa.googleapis.com` causes you to be unable to download/install the Chrome browser, presumably on Windows. Haven't had time to check this yet, so if anyone has time ...

**Comment by hagezi** [comment]
@Retold3202 I have tested it under Windows, I cannot reproduce the problem.

**Comment by ChaseKnowlden** [comment]
Can't reproduce chrome download issue on Windows.

**Comment by haroondilshad** [comment]
Learnt the hard way that blocking optimizationguide-pa.googleapis.com disables critical ML model download functionality so basic stuff like translation does not work any longer. Following models can not be downloaded if you have this disabled

![Screenshot 2024-08-30 at 12 28 11 AM](https://github.com/user-attachments/assets/0683f149-ca46-41bd-9106-5a1e27491094)
![Screenshot 2024-08-30 at 12 28 19 AM](https://github.com/user-attachments/assets/495e1593-3d31-4d6d-bab9-dabf652c5c79)

**Comment by hagezi** [comment]
Thanks @haroondilshad

**Comment by ExtRIELICi** [comment]
> ```
> # Google Wallet - breaks adding PKPASS files
> auditrecording-pa.googleapis.com
> ```
> 
> ```
> # Needed for additional Google Assistant downloads on Android
> geller-pa.googleapis.com
> ```

I don’t use an Android device, but these two domains frequently appear in my logs. Could we consider adding them to Ultimate only? They’re already blocked for me by 1Hosts Pro, and I haven’t experienced any issues. For context, the only Google services I use are Google Maps and the YouTube app, I don’t even use their search engine.

<img width="333" alt="image" src="https://github.com/user-attachments/assets/e7de83d1-22bb-46d9-83f3-4fc357bb3891" />

<img width="276" alt="image" src="https://github.com/user-attachments/assets/560e389d-e28e-4a10-aff0-3cca689c5e7c" />

**Comment by celenityy** [comment]
I second this: While not called as often as what @ExtRIELICi is seeing, I do see both `auditrecording-pa.googleapis.com` & `geller-pa.googleapis.com` on a regular basis:

![image](https://github.com/user-attachments/assets/cbf54be3-5268-46b9-b3dd-e722d0734ac7)

![image](https://github.com/user-attachments/assets/3d9dd42c-5cbf-4ec5-97d9-7566252b244a)

Ultimate is supposed to be an aggressive list - not blocking these domains due to niche issues isn't the right move IMO - for the same reasons we still block Facebook's `graph` domains & some of the Microsoft domains on `Ultimate` despite causing breakage.

**Comment by StrangePeanut** [comment]
I know this isn't very helpful as I don't know which exact domain or domains are responsible, but I can confirm that Google Assistant and Gemini are still being affected on first time use today. Everything seems to be fine after allowing the device to connect once.

Lists in use

- HaGeZiMultiPro
- HaGeZiTIF

**Comment by olumolu** [comment]
I also use this two
But i never faced any such issues with gemini or g assistant.

**Comment by celenityy** [comment]
@StrangePeanut What domains are you seeing blocked around the time you notice the issues? My *suspicion* is `geller-pa.googleapis.com` - in which case, I’d be curious to hear if unblocking it fixes your problem.

**Comment by hagezi** [comment]
I can't reproduce this, `geller-pa.googleapis.com` is only blocked in Ultimate. I also don't know what else could cause this behavior, without logs we won't get any further here. Maybe it was just a “network hiccup” or a DNS problem.

**Comment by StrangePeanut** [comment]
It took some effort but I can now share with certainty that it's `footprints-pa.googleapis.com`. No Google Assistant and Gemini breakage occur as long as this domain is whitelisted.

**Comment by hagezi** [comment]
Thanks @StrangePeanut, I will only block footprints-pa.googleapis.com in Ultimate.

**Comment by hagezi** [comment]
@Punisher-Username1:
https://github.com/hagezi/dns-blocklists/issues/2900#issuecomment-2567697723
https://github.com/hagezi/dns-blocklists/issues/2900#issuecomment-2571492733
https://github.com/hagezi/dns-blocklists/issues/2900#issuecomment-2571513897

**Comment by celenityy** [comment]
@ExtRIELICi 

ChatGPT is incorrect. `metric.gstatic.com` is used for DNS over TLS connectivity checks on Android. You can see it in AOSP's source code [here](https://cs.android.com/search?q=metric.gstatic.com&sq=&ss=android%2Fplatform%2Fsuperproject%2Fmain). I'm not aware of it being used for tracking/analytics.

**Comment by hagezi** [comment]
ref: Blocking `semanticlocation-pa.googleapis.com` possibly breaks Google Maps
https://github.com/hagezi/dns-blocklists/issues/4945

**Comment by dandv** [comment]
Any information about `peoplestackwebexperiments-pa.clients6.google.com`?

I've been blocking it from Spotify for a few days, apparently with no loss of functionality. Saw it accessed by various Google services (not just photos) in the browser.

**Comment by hagezi** [comment]
https://www.perplexity.ai/search/peoplestackwebexperiments-pa-c-ws2ZbdPGS62SH1mdQ_UKrg#1

**Comment by dandv** [comment]
@hagezi I get "This thread does not exist" when I follow that link.

When I searched for the domain name, Perplexity regurgitated what we already know:

```
peoplestackwebexperiments-pa.clients6.google.com is a subdomain of Google that is related to Google Photos Face Grouping[1](https://www.reddit.com/r/privacy/comments/13kx5wn/what_is/)[7](https://github.com/llacb47/mischosts/issues/19). DNS resolution of peoplestackwebexperiments-pa.clients6.google.com points to 142.250.99.95, which is located in Mountain View, California[3](https://domain.glass/peoplestackwebexperiments-pa.clients6.google.com).

Although it is related to Google Photos, some users have reported seeing calls to this subdomain even when they don't use Google Photos[1](https://www.reddit.com/r/privacy/comments/13kx5wn/what_is/).

According to domain.glass, peoplestackwebexperiments-pa.clients6.google.com scored 867 on 2020-11-01, based on global DNS query popularity[3](https://domain.glass/peoplestackwebexperiments-pa.clients6.google.com).
```

**Comment by hagezi** [comment]
Your device is likely calling peoplestackwebexperiments-pa.clients6.google.com as part of Google's services, specifically related to Google Photos Face Grouping. This feature uses facial recognition to organize photos, even if you do not actively use Google Photos. The subdomain may be accessed by background processes linked to Google Play Services or other pre-installed Google apps that rely on shared infrastructure for functionality like syncing, telemetry, or AI-based features [1](https://www.reddit.com/r/privacy/comments/13kx5wn/what_is/) [2](https://forum.f-droid.org/t/usage-of-urls-after-disble-most-google-apps/16427) [7](https://pulsedive.com/ioc/peoplestackwebexperiments-pa.clients6.google.com).

If you have disabled most Google apps or do not use Google Photos, these calls could still occur due to residual dependencies in the operating system or apps that interact with Google's services. Blocking this domain may not disrupt your device's core functions but could affect specific features tied to Google's ecosystem [2](https://forum.f-droid.org/t/usage-of-urls-after-disble-most-google-apps/16427) [5](https://github.com/nickspaargaren/no-google/blob/master/pihole-google-unbound.conf).

**Comment by celenityy** [comment]
`peoplestackwebexperiments-pa.clients6.google.com` specifically appears to be related to some kind of A/B testing/experimentation *(likely related to Google Photos/Face Groups as mentioned above)*. In my testing, I noticed this specifically called from `drive.google.com`:

`https://peoplestackwebexperiments-pa.clients6.google.com/$rpc/peoplestackwebexperiments.PeopleStackExperimentsService/GetExperimentFlags`

That surprises me to hear it’s being called by Spotify, I’m not sure why third party services would be using it. Regardless, I would be in favor of blocking `peoplestackwebexperiments-pa.clients6.google.com` under the assumption it doesn’t cause breakage. I don’t use Google Drive/Photos/etc. personally, so I can’t really comment there.

**Comment by xRuffKez** [comment]
No blocking! ```myphonenumbers-pa.googleapis.com```  Likely handles phone number verification or management for Google services.
No blocking! ```phonedeviceverification-pa.googleapis.com```  Probably used for verifying phone devices linked to Google accounts.
```performanceparameters.googleapis.com```  Might collect or manage performance metrics for Google apps or services.
```searchlabspartnerservice-pa.googleapis.com``` Could be related to experimental or partner-based search features.
```semanticlocation-pa.googleapis.com``` Likely deals with location-based services using semantic analysis (e.g., identifying meaningful locations).

**Comment by celenityy** [comment]
I noticed `searchlabspartnerservice-pa.googleapis.com` called from the `Google` app on Android, it appears to be related to ['Search Labs/Experiments'](https://support.google.com/websearch/answer/13572013), which looks like more A/B testing/experimentation. The fact this functionality [is unavailable in Europe due to local regulations](https://support.google.com/websearch/answer/14184960#zippy=%2Ceurope-the-middle-east-africa) speaks volumes... I think this is definitely worth blocking.

`semanticlocation-pa.googleapis.com` was blocked previously, [but was found to break Google Maps on Android](https://github.com/hagezi/dns-blocklists/issues/4945), so it was removed.

It looks like `myphonenumbers-pa.googleapis.com` [is called by Google Play Services](https://pastebin.com/H2W9v6ad) on Android. My impression from [skimming AOSP's source code](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/telephony/common/com/google/android/mms/pdu/PduPersister.java;l=1450?q=myphonenumbers) is that `myphonenumbers-pa.googleapis.com` is being used to send the user's phone numbers *(detected via their SIM card(s))* to Google. I don't see why this would be necessary/what legitimate functionality it provides, so I think it might be worth blocking - maybe just on `Ultimate` for now though to see if anyone reports any issues. I'll also add this to my BadBlock lists.

`phonedeviceverification-pa.googleapis.com` is odd - it appears to be called by Google Play Services on Android *(I've specifically found: `phonedeviceverification-pa.googleapis.com/google.internal.communications.phonedeviceverification.v1.PhoneDeviceVerification/GetConsent)*, but that's about all I can find on it. I also discovered `phonedeviceverification-pa-prod.sandbox.googleapis.com`, which appears to be similar and also used for this same mystery functionality. This will need more research and testing, I'll probably block both of these on my network and report back if I hear any complaints/issues/etc...

`performanceparameters.googleapis.com` appears to be used for [collecting performance insights in games](https://android-developers.googleblog.com/2020/06/performance-insights-for-games.html), so I think it should be blocked. For reference, I've had this blocked on BadBlock for several months and haven't received any issues.

**Comment by hagezi** [comment]
Thanks @celenityy, I will add 

```
searchlabspartnerservice-pa.googleapis.com
myphonenumbers-pa.googleapis.com
performanceparameters.googleapis.com
```

to the Ultimate.

**Comment by dandv** [comment]
> Your device is likely calling peoplestackwebexperiments-pa.clients6.google.com as part of Google's services

My device is a laptop running Fedora, and I don't use any Google applications outside of a browser. I do develop code that make API calls to Firebase and Google Auth, but that's it.

> That surprises me to hear it’s being called by Spotify

This *might* be an [issue with the firewall I'm using](https://github.com/evilsocket/opensnitch/issues/929) - mis-identifying the application.

**Comment by dandv** [comment]
> That surprises me to hear it’s being called by Spotify, I’m not sure why third party services would be using it.

@celenityy: Spotify [uses the Chromium](https://www.spotify.com/us/opensource/) Embedded Framework, so it might be Chromium phoning home.

**Comment by olumolu** [comment]
hi what this three domains do
proactivebackend-pa.googleapis.com
firebaseremoteconfigrealtime.googleapis.com
firebaseremoteconfig.googleapis.com

**Comment by celenityy** [comment]
@olumolu 

`proactivebackend-pa.googleapis.com` is used for Google Assistant/Gemini on Android. I'd recommend blocking it if you don't use/care about this functionality *(due to the severe privacy implications...)*, but I doubt it's something hagezi would block for the masses.

`firebaseremoteconfig.googleapis.com` & `firebaseremoteconfigrealtime.googleapis.com` *(as well as `baseremoteconfig.googleapis.com`)* are used for [A/B Testing/Experimentation](https://firebase.google.com/products/remote-config) and ["personalization"...](https://firebase.google.com/case-studies/halfbrick-personalization). I'd be in favor of blocking these.

**Comment by hagezi** [comment]
You have to be careful when blocking certain firebase domains, as these are required for push notifications in countless apps (cloud messaging). At least firebaseinstallations and firebaseremoteconfig are required for this.

**Comment by olumolu** [comment]
After blocking my self proactivebackend-pa.googleapis.com
This domain i can still use gemini i really dont think it is for ai like gemini

**Comment by hagezi** [comment]
@ExtRIELICi 

![Image](https://github.com/user-attachments/assets/85d25f92-5be3-479f-8513-3c075555e2eb)

**Comment by celenityy** [comment]
@ExtRIELICi 

From my [push notifications whitelist](https://codeberg.org/celenity/BadBlock#push-notifications-whitelist):

```sh
alt1-mtalk.google.com
alt2-mtalk.google.com
alt3-mtalk.google.com
alt4-mtalk.google.com
alt5-mtalk.google.com
alt6-mtalk.google.com
alt7-mtalk.google.com
alt8-mtalk.google.com
android.googleapis.com
fcmconnection.googleapis.com
fcm.googleapis.com
fcm-xmpp.googleapis.com
firebaseinappmessaging.googleapis.com
firebaseperusertopics-pa.googleapis.com
gcm-http.googleapis.com
gcm-xmpp.googleapis.com
instantmessaging.googleapis.com
instantmessaging-pa-eu.googleapis.com
instantmessaging-pa.googleapis.com
instantmessaging-pa-jms-eu.googleapis.com
instantmessaging-pa-jms-us.googleapis.com
instantmessaging-pa-us.googleapis.com
mobile-gtalk4.l.google.com
mobile-gtalk.l.google.com
mtalk4.google.com
mtalk-dev.google.com
mtalk.google.com
mtalk-staging.google.com
notifications.google.com
notifications-pa.googleapis.com
```

Some have also claimed `firebaseinstallations.googleapis.com` is responsible for push notifications, though I’m unable to confirm that’s the case based off testing and research.

**Comment by rossengeorgiev** [comment]
@celenityy I've seen applications not able to deliver push notification when `firebaseinstallations.googleapis.com` is blocked during setup. I believe the domain only needs to be whitelisted at application setup time when it "registers" for push notifications, after that it can remain blocked.

To reproduce, try force stopping & clear data on the gmail app, then open the app and access your gmail. If `firebaseinstallations.googleapis.com` is blocked during that initial setup, you won't get push notifications for new emails, if it was allowed you would. Tested on fully updated Android 15.

**Comment by AnimMouse** [comment]
It seems like `auditrecording-pa.googleapis.com` is also used for Google Find Hub (formerly Find My Device) trackers. I can't add my Find Hub trackers if that domain is blocked. I also can't share my trackers to other accounts if that domain is blocked.

**Comment by xRuffKez** [comment]
@AnimMouse that could be my fix! I tried to ring my missing phone over find my mobile but it was not able to connect to my phone.

**Comment by hagezi** [comment]
@xRuffKez only blocked in Ultimate, reason fro blocking: https://github.com/hagezi/dns-blocklists/issues/2900#issuecomment-2571513897

**Comment by xRuffKez** [comment]
@hagezi Oh... well i have to figure it out further more. Thanks

**Comment by IsHacker003** [comment]
> Some have also claimed `firebaseinstallations.googleapis.com` is responsible for push notifications, though I’m unable to confirm that’s the case based off testing and research.

It is used for Google Cloud Messaging. The Telegram X app is not able to receive messages in background when it's blocked, and shows an error in the notification settings asking to whitelist this domain.

**Comment by hagezi** [comment]
Yes, and this problem exists in many other apps. I had this in the F1TV app, for example. The push notifications could also not be configured if this domain was blocked.

**Comment by hagezi** [comment]
https://github.com/hagezi/dns-blocklists/issues/7553

**Comment by olumolu** [comment]
`photosdata-pa.googleapis.com`
Do you have any idea what this one is used for i have blocked 2days ago and haven't noticed anything broken yet?.

**Comment by hagezi** [comment]
`photosdata-pa.googleapis.com` is used by Google Photos background services. This domain is involved in the background synchronization and backup of photos and videos to Google Photos storage.

**Comment by olumolu** [comment]
> `photosdata-pa.googleapis.com` is used by Google Photos background services. This domain is involved in the background synchronization and backup of photos and videos to Google Photos storage.

I see i use ente now and before that i used to keep offline so i never saw any issues..
Thanks

**Comment by Docga1835** [comment]
I wanted to add in case any one runs into this like me, since `volatile-pa.googleapis.com` was mentioned earlier in this thread, that blacklisting this domain breaks the ability to use speech-to-text with the Gboard keyboard on Galaxy smart watch. Unblocking it restored functionality for me.

**Comment by hagezi** [comment]
Thanks @Docga1835, I'll add that to the known issues for Ultimate.

**Comment by hagezi** [comment]
https://github.com/hagezi/dns-blocklists/issues/7981

**Comment by hagezi** [comment]
#8272

**Comment by Alvoracz** [comment]
As Ghost said [here](https://github.com/hagezi/dns-blocklists/issues/2900#issuecomment-2590327413) unblocking voilatile-pa.googleapis.com helps with maps, because for me, it provides network based location.

**Comment by hagezi** [comment]
A summary of a statement I received from a Tech Lead on the Audit Recording Infrastructure team at Google:

> Critical function of the endpoint `auditrecording-pa.googleapis.com` 
> 
> This enpoint essential for Google services to create durable audit records of user consents and other sensitive user interactions. This is often necessary for regulatory compliance (like GDPR) and to ensure transparency and user control. It is not related to Google's Cloud Audit Logging API (https://cloud.google.com/logging/docs/audit), which is a separate service.
> 
> It's important to note that these audit records are strictly for compliance and legal investigation purposes; this data is not used and can't be used for other purposes like personalized ads.
> 
> The impact of blocking this domain can be significant:
> 
> 1. Essential Functionality Breaks: For mobile applications blocking this domain can prevent critical actions. If a product calls ARI in a blocking manner to record mandatory consent, the function call will fail. This means users might be unable to save new preferences, settings, or complete other actions that require an audit trail. We've seen reports of this affecting services like Google Maps and Google Find Hub, where users are unable to add or share trackers.
> 
> 2. Incomplete Audit Trails: Even for non-blocking calls, the blockage prevents client-side audit records, often containing important UI context, from reaching our servers.
> 
> Blocking `auditrecording-pa.googleapis.com` inadvertently harms users by potentially breaking service functionality and compromising the integrity of audit trails for their consents and choices. It is in the best interest of users to have this domain accessible to ensure their Google services function correctly and their interactions are properly recorded.

The domain is currently blocked only in the Ultimate list and will be removed in the next release.

Issues reported so far regarding the domain:

```
# Google Wallet - breaks adding PKPASS files
# Find My Device - breaks adding/sharing trackers
```

**Comment by Efreak** [comment]
Sort of off-topic for this repository, but on topic for the thread, so I'm necroing (sorry!)

Is anyone aware of a list of what different Google subdomains are used for? I've been searching for the past while and only managed to find fragmented information in specific subdomain-related questions on Reddit and GitHub issues. I'm assuming the common -pa suffix means something, but it's incredibly hard to find given the initial `-` and only two more letters making it impossible to search for. The rest of it can only be guessed at based on the name (as seen in this thread). It would be really nice to have a list like the [universal debloater alliance](https://github.com/Universal-Debloater-Alliance/universal-android-debloater-next-generation/) has.

It would actually be useful to have such a list for more than just Google; Samsung's domains are even more opaquely named, and other web services use multiple domains and subdomains as well.
